socialhome/content/models.py

```diff
@@ -118,6 +118,8 @@ def save(self, author=None, *args, **kwargs):
             # New with author, set a GUID and author
             self.guid = uuid4()
             self.author = author
+            if self.pinned:
```

Comment:
Hmm this whole `elif author:` is kinda weird, had to dig to find out why it isn't just an `else:`. Anyway, your code should run even if no author. So maybe restructure like:

    if self.pk:
        ...
    else:
        if self.pinned:
            ...
        if author:
            ....